src/monster.cpp

```diff
@@ -2633,17 +2633,19 @@ void monster::process_effects()
 
     Character &player_character = get_player_character();
     //If this monster has the ability to heal in combat, do it now.
-    const int healed_amount = heal( type->regenerates );
-    if( healed_amount > 0 && one_in( 2 ) ) {
-        std::string healing_format_string;
-        if( healed_amount >= 50 ) {
-            healing_format_string = _( "The %s is visibly regenerating!" );
-        } else if( healed_amount >= 10 ) {
-            healing_format_string = _( "The %s seems a little healthier." );
-        } else {
-            healing_format_string = _( "The %s is healing slowly." );
+    if( type->regenerates && !has_effect( effect_onfire ) ) {
```

Comment:
This needs to be a property any effect can opt into, not hardcoded to a specific effect id. I'd suggest you add it as a flag.

Reply:
If I understand your suggestion, add a 'prevents_regeneration' flag to the 'onfire' effect, and change this function to check for any effect with that flag?

Reply:
Yes.